src/app/services/ckan.service.ts

```diff
@@ -2,70 +2,96 @@ import { Injectable } from '@angular/core';
 
-  searchDatasets(query: string, start: number = 0, rows: number = 12): Observable<{ results: any[], count: number }> {
-    const url = `${environment.backendUrl}/api/3/action/package_search?q=${encodeURIComponent(query)}&start=${start}&rows=${rows}`;
+  searchDatasets(
+    query: string,
+    filter: string = '',
+    start: number = 0,
+    rows: number = 12
+  ): Observable<{ results: PartialDataset[]; count: number }> {
+    const url = `${environment.backendUrl}/api/action/package_search?q=${encodeURIComponent(
+      query
+    )}&fq=${encodeURIComponent(filter)}&start=${start}&rows=${rows}`;
     return this.http.get<any>(url).pipe(
-      map(response => {
-
+      map((response) => {
         const items = response.result.results.map((item: any) => ({
           id: item.id,
           title: item.title,
           description: item.notes,
           modified: item.metadata_modified,
-          organizationName: item.organization.title,
-          publisher_name: item.publisher_name
+          organization: item.organization.title,
+          publisher_name: item.publisher_name,
+          tags: item.tags.map((tag: { name: string }) => tag.name),
+          theme: item.theme[0]
+            ?.slice(2, -2)
+            .replaceAll(/https?:\/\//g, '')
+            .replaceAll('"', '')
+            .replaceAll(' ', '')
+            .split(','),
+          format: item.resources[0]?.format.toLowerCase(),
         }));
 
         return {
           results: items,
-          count: response.result.count
+          count: response.result.count,
         };
       }),
-      catchError(error => {
+      catchError((error) => {
         console.error(error);
         return of({ results: [], count: 0 });
       })
     );
   }
 
   getSchemingPackageShow(id: string): Observable<Dataset> {
-    return this.http.get<Dataset>(`${environment.backendUrl}/api/3/action/scheming_package_show?type=dataset&id=${id}`);
+    return this.http.get<Dataset>(
+      `${environment.backendUrl}/api/action/scheming_package_show?type=dataset&id=${id}`
+    );
   }
 
-  getCatalogueDetails(): Observable<any> {
-    const urls = [...this.ckanToDcat.keys()].map(item => `${environment.backendUrl}/api/3/action/${item}_list`);
+  getCatalogueDetails(): Observable<CatalogueDetail[]> {
+    const urls = [...CkanService.ckanToDcat.keys()].map(
+      (item) => `${environment.backendUrl}/api/3/action/${item}_list`
+    );
 
     return from(urls).pipe(
-      mergeMap(url => this.getCatalogueDetail(url)),
-      toArray(),
+      mergeMap((url) => this.getCatalogueDetail(url)),
+      toArray()
     );
   }
 
-  getCatalogueDetail(url: string): Observable<any> {
-    let itemCategory: string = this.ckanToDcat.get(url.split("/").pop()?.split("_")?.[0] ?? "") ?? "";
+  private getCatalogueDetail(url: string): Observable<CatalogueDetail> {
+    let itemCategory: string = CkanService.ckanToDcat.get(url.split('/').pop()!.split('_')[0])!;
```

Comment:
You parse to URL, to parse it back to key here. Can you simply passa both key and url as args for this function? Or parse to URL inside the function?